test.sh

```diff
@@ -0,0 +1,21 @@
+echo “$(tput setaf 3) SHIELDCONTRACTS” 
+SOLFILES=$(find . -type f -name "*Shield*.sol" -maxdepth 4 -mindepth 4)
+solarray=($SOLFILES)
+for solelement in "${solarray[@]}"
+do
+    DIR="$(dirname "${solelement}")"
+    solelement="${solelement:2}"
+    echo “$(tput setaf 7) $solelement compiling” 
+    solcjs --abi -o $DIR --include-path node_modules/ --base-path . $solelement || echo “$(tput setaf 1) $solelement failed”
+done
+
+echo “$(tput setaf 3) CIRCUITS” 
+ZOKFILES=$(find . -type f -name "*.zok" -maxdepth 4 -mindepth 4)
```

Comment:
This all works very cleanly! I have a suggestion - is it possible to skip circuits in the `common` folders? We know they work, and some don't compile alone since they have no `main` function, which is fine but looks like an error

Reply:
Done. Tests are performed only in zapps folder